File: capstone-projects/cafe-management-program/cpp/Project_Nhom04_Cpp/Account.cpp
#include "Library.h"

std::string Account::getsUsername() {
	return sUsername;
}

void Account::setsUsername(std::string sUsername) {
	this->sUsername = sUsername;
}

std::string Account::getsPassword() {
	return sPassword;
}

void Account::setsPassword(std::string sPassword) {
	this->sPassword = sPassword;
}

Account::Account() {
}

Account::~Account() {
}

Account::Account(std::string username, std::string password) {
	this->sUsername = username;
	this->sPassword = password;
}

//Input
void Account::InputUsername() {
	printf("Username: ");
	this->sUsername = toLower(InputString());
}

void Account::InputPassword() {
	printf("Password: ");
	this->sPassword = InputString();
}

void Account::Input() {
	InputUsername();
	InputPassword();
}

void Account::AddUsername() {
	printf("Username: ");
	std::string username = toLower(InputString());
	while (IsExist(username)) {
		printf("\n\t\tUsername Is Used. Please Type Another Username!\n");
		printf("Username: ");
		username = toLower(InputString());
	}
	this->setsUsername(username);
}

void Account::AddAccount() {
	AddUsername();
	InputPassword();
}

//Methods
bool IsExist(std::string username) {
	for (int i = 0; i < lStaffs.size(); i++) {
		if (toLower(username) == toLower(lStaffs.at(i).getaAccount().getsUsername()))
			return true;
	}
	return false;
}

bool IsExist(Account account) {
	for (int i = 0; i < lStaffs.size(); i++) {
		if (IsEqual(account, lStaffs.at(i).getaAccount()))
			return true;
	}
	return false;
}

void AccessLogin(Account login) {
	for (int i = 0; i < lStaffs.size(); i++) {
		if (IsEqual(login, lStaffs.at(i).getaAccount())) {
			if (!IsDate(nDate.getiDay(), nDate.getiMonth(), nDate.getiYear()))
				InputDate(lStaffs.at(i).getsName(), lStaffs.at(i).getsID());
			if (lStaffs.at(i).getsType() == "Staff")
				lStaffs.at(i).LoginStaff();
			else
				lStaffs.at(i).Login();
		}
	}
}

//Operator
bool IsEqual(Account a, Account b) {
	return (toLower(a.getsUsername()) == toLower(b.getsUsername())) && (a.getsPassword() == b.getsPassword());
}

bool IsNotEqual(Account a, Account b) {
	return !(IsEqual(a, b));
}

File: capstone-projects/cafe-management-program/java/src/Account.java
import java.io.IOException;

public class Account {
    //Fields
    protected String sUsername;
    protected String sPassword;

    //Propeties
    public String getsUsername() {
        return sUsername;
    }

    public void setsUsername(String sUsername) {
        this.sUsername = sUsername;
    }

    public String getsPassword() {
        return sPassword;
    }

    public void setsPassword(String sPassword) {
        this.sPassword = sPassword;
    }

    //Constructors
    public Account() {
    }

    public Account(String username, String password) {
        this.sUsername = username;
        this.sPassword = password;
    }

    //Input
    public void InputUsername() {
        System.out.print("Username: ");
        this.sUsername = Main.InputString().toLowerCase();
    }

    public void InputPassword() {
        System.out.print("Password: ");
        this.sPassword = Main.InputString();
    }

    public void Input() {
        InputUsername();
        InputPassword();
    }

    public void AddUsername() {
        System.out.print("Username: ");
        String username = Main.InputString().toLowerCase();
        while (IsExist(username)) {
            System.out.println("\n\t\tUsername Is Used. Please Type Another Username!\n");
            System.out.print("Username: ");
            username = Main.InputString().toLowerCase();
        }
        this.setsUsername(username);
    }

    public void AddAccount() {
        AddUsername();
        InputPassword();
    }

    //Methods
    static public boolean IsExist(String username) {
        for (int i = 0; i < Cafe.getlStaffs().size(); i++) {
            if (username.equalsIgnoreCase(Cafe.getlStaffs().get(i).getaAccount().getsUsername()))
                return true;
        }
        return false;
    }

    static public boolean IsExist(Account account) {
        for (int i = 0; i < Cafe.getlStaffs().size(); i++) {
            if (IsEqual(account, Cafe.getlStaffs().get(i).getaAccount()))
                return true;
        }
        return false;
    }

    static public void AccessLogin(Account login) throws IOException {
        for (int i = 0; i < Cafe.getlStaffs().size(); i++) {
            if (IsEqual(login, Cafe.getlStaffs().get(i).getaAccount())) {
                if (!Date.IsDate(Main.dDate.getiDay(), Main.dDate.getiMonth(), Main.dDate.getiYear()))
                    Main.InputDate(Cafe.getlStaffs().get(i).getsName(), Cafe.getlStaffs().get(i).getsID());
                Cafe.getlStaffs().get(i).Login();
            }
        }
    }

    //Operator
    public static boolean IsEqual(Account a, Account b) {
        return (a.getsUsername().equalsIgnoreCase(b.getsUsername())) && (a.getsPassword().equals(b.getsPassword()));
    }

    public static boolean IsNotEqual(Account a, Account b) {
        return !(IsEqual(a, b));
    }
}
